Cargo.toml

```diff
@@ -10,8 +10,19 @@ categories = ["network-programming"]
 readme = "README.md"
 documentation = "https://docs.rs/ipnet"
 
+[features]
+default = []
+# Implements "serde::Serialize" and "serde::Deserialize".
+serde = ["serde_crate"]
```

Comment:
Just by making the crate optional (`optional = true`, as already done) you can enable it using a feature flag of the crate name itself.

So then this line is not needed and `serde` feature can just be used immediately. 
For more info: https://stackoverflow.com/a/39759592/2037998

This way some of the changes in `src/ipnet_serde.rs` are no longer needed to. That will make the PR a bit simpler.

Reply:
I've just pushed something, take a look. Let me know if that is good

Reply:
I don't see the changes anywhere. Where did you push them?

Reply:
I went ahead and collapsed this, avoiding the renaming.

`cargo build/test --features serde` and `cargo build/test --features json` are both happy.